cmd/jsonnetdeps/cmd.go

```diff
@@ -0,0 +1,252 @@
+			}
+			conf.outputFile = outputFile
+		} else if arg == "-J" || arg == "--jpath" {
+			dir := cmd.NextArg(&i, args)
+			if len(dir) == 0 {
+				return processArgsStatusFailure, fmt.Errorf("-J argument was empty string")
+			}
+			if dir[len(dir)-1] != '/' {
+				dir += "/"
+			}
+			conf.jPaths = append(conf.jPaths, dir)
+		} else if arg == "-s" || arg == "--max-stack" {
+			l := safeStrToInt(cmd.NextArg(&i, args))
+			if l < 1 {
+				return processArgsStatusFailure, fmt.Errorf("invalid --max-stack value: %d", l)
+			}
+			vm.MaxStack = l
+		} else if arg == "-t" || arg == "--max-trace" {
+			l := safeStrToInt(cmd.NextArg(&i, args))
+			if l < 0 {
+				return processArgsStatusFailure, fmt.Errorf("invalid --max-trace value: %d", l)
+			}
+			vm.ErrorFormatter.SetMaxStackTraceSize(l)
+		} else if arg == "--" {
+			// All subsequent args are not options.
+			i++
+			for ; i < len(args); i++ {
+				remainingArgs = append(remainingArgs, args[i])
+			}
+			break
+		} else if len(arg) > 1 && arg[0] == '-' {
+			return processArgsStatusFailure, fmt.Errorf("unrecognized argument: %s", arg)
+		} else {
+			remainingArgs = append(remainingArgs, arg)
+		}
+	}
+
+	want := "filename"
+	if conf.filenameIsCode {
+		want = "code"
+	}
+	if len(remainingArgs) == 0 {
+		return processArgsStatusFailureUsage, fmt.Errorf("must give %s", want)
+	}
+
+	multipleFilesAllowed := false
+	if !multipleFilesAllowed {
+		if len(remainingArgs) > 1 {
+			return processArgsStatusFailure, fmt.Errorf("only one %s is allowed", want)
+		}
+	}
+
+	conf.inputFiles = remainingArgs
+	return processArgsStatusContinue, nil
+}
+
+func writeDependencies(dependencies []string, outputFile string) (err error) {
+	var f *os.File
+
+	if outputFile == "" {
+		f = os.Stdout
+	} else {
+		f, err = os.Create(outputFile)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if ferr := f.Close(); ferr != nil {
+				err = ferr
+			}
+		}()
+	}
+
+	output := strings.Join(dependencies, "\n") + "\n"
+	_, err = f.WriteString(output)
+	if err != nil {
+		return err
+	}
+
+	return
+}
+
+func main() {
+	cmd.StartCPUProfile()
+	defer cmd.StopCPUProfile()
+
+	vm := jsonnet.MakeVM()
+	vm.ErrorFormatter.SetColorFormatter(color.New(color.FgRed).Fprintf)
+
+	conf := config{}
+	jsonnetPath := filepath.SplitList(os.Getenv("JSONNET_PATH"))
+	for i := len(jsonnetPath) - 1; i >= 0; i-- {
+		conf.jPaths = append(conf.jPaths, jsonnetPath[i])
+	}
+
+	status, err := processArgs(os.Args[1:], &conf, vm)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "ERROR: "+err.Error())
+	}
+	switch status {
+	case processArgsStatusContinue:
+		break
+	case processArgsStatusSuccessUsage:
+		usage(os.Stdout)
+		os.Exit(0)
+	case processArgsStatusFailureUsage:
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "")
+		}
+		usage(os.Stderr)
+		os.Exit(1)
+	case processArgsStatusSuccess:
+		os.Exit(0)
+	case processArgsStatusFailure:
+		os.Exit(1)
+	}
+
+	vm.Importer(&jsonnet.FileImporter{JPaths: conf.jPaths})
+
+	if len(conf.inputFiles) != 1 {
```

Comment:
Can you change the representation so that conf can only have one file (i.e. `inputFile []string` -> `inputFile string`)? It's good to make invalid states unrepresentable.

Alternatively you can actually support passing multiple files. It would then produce union of depeendency sets of the passed files.

Reply:
Updated to handle multiple input files.